### PR TITLE
add support for docker-for-windows new bind serialization format

### PR DESCRIPTION
**Pull request by roastario**
docker-for-windows has recently changed the way it presents bind information (see: https://github.com/docker-java/docker-java/issues/1462) this PR fixes this client + docker-for-windows.

### CODE REVIEW

**Comment by survivant** [comment]
the fix is it include with : build edge 2.4.1.0 (48583) ?

**Review by leCandas — APPROVED** [review]

**Comment by survivant** [comment]
will it be release soon ?  still present in docker edge 4.2.x.x

**Comment by survivant** [comment]
Issue still present in Docker Desktop version 2.5.0.0.

**Comment by survivant** [comment]
@bsideup can you give a little code review and approve it please.  thanks

File: docker-java-transport-zerodep/pom.xml [hunk 1]
```diff
@@ -48,6 +48,7 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-shade-plugin</artifactId>
+				<version>3.2.4</version>
```
Comment (bsideup) [inline docker-java-transport-zerodep/pom.xml:51]: could you please revert this unrelated change?

File: docker-java-api/src/main/java/com/github/dockerjava/api/model/Bind.java [hunk 2]
```diff
@@ -99,36 +99,39 @@ public static Bind parse(String serialized) {
...
-                        propagationMode = PropagationMode.SLAVE;
-                    } else if (PropagationMode.PRIVATE.toString().equals(p)) {
-                        propagationMode = PropagationMode.PRIVATE;
-                    } else {
-                        seMode = SELContext.fromString(p);
-                    }
+                case 2: {
+                    return new Bind(parts[0], new Volume(parts[1]));
                 }
+                case 4: {
```
Comment (bsideup) [inline docker-java-api/src/main/java/com/github/dockerjava/api/model/Bind.java:105]: please put it after the 3rd case

**Comment by bsideup** [comment]
@survivant thanks for the ping! The implementation looks okay, I just requested a couple of minor changes, but should be good to go once fixed

File: docker-java-api/src/main/java/com/github/dockerjava/api/model/Bind.java (hunk 2 above)
Comment (alanwilkie-finocomp) [reply docker-java-api/src/main/java/com/github/dockerjava/api/model/Bind.java:105]: Presumably it's handled here to get "fall through" behaviour after consolidating the first two parts. Would it be better to move this to an `if (parts.length == 4)` block before the `switch`?

Also, does Docker for Windows ever present a variant without the access mode parts (i.e. is there a 3-part version like `C:/foo:/bar`) that `case 3` wouldn't handle?

**Comment by survivant** [comment]
@roastario @bsideup just a ping so the bot doesn't close it.  Still need it be to able to use it with testcontainer and recent version of Docker.

**Comment by roastario** [comment]
Apologies, I'd accidentally snoozed notifications for this repo. 

Will look at the comments.

**Comment by survivant** [comment]
just a ping.

**Review by bsideup — APPROVED** [review]

**Comment by bsideup** [comment]
@roastario I just applied a couple of changes myself and will merge once CI is green.
